lib/temporal/client/serializer/complete_workflow.rb

```diff
@@ -10,7 +9,7 @@ def to_proto
             command_type: Temporal::Api::Enums::V1::CommandType::COMMAND_TYPE_COMPLETE_WORKFLOW_EXECUTION,
             complete_workflow_execution_command_attributes:
               Temporal::Api::Decision::V1::CompleteWorkflowExecutionCommandAttributes.new(
-                result: Payload.new(object.result).to_proto
+                result: object.result.nil? ? nil : Temporal.configuration.converter.to_payloads([object.result])
```

Comment:
I wonder why we need this exception? If we always expect the payload data to be an array we can probably avoid special-casing `nil` results. Or am I missing something?

Reply:
`Payloads` with one `Payload` which is `nil`, is not the same as having `Payloads` itself as `nil` which is what this does. This represents the difference between `return []` and `return nil` from a workflow.